Automattic/wp-calypso
# Refactor site migration credentials form

Related to https://github.com/Automattic/wp-calypso/issues/94236

## Proposed Changes

Refactored the site migration credentials form by breaking it down into smaller components and hooks.

Also, now we show individual error message for userid and password because otherwise messages applicable for userid gets shown below the password field, which looks confusing.

<img width="593" alt="Screenshot 2024-09-11 at 4 13 11 PM" src="https://github.com/user-attachments/assets/5eba1294-e2c1-4dd6-9f20-585c91d2dc00">

## Why are these changes being made?


* Because the component became too big and hard to read

## Testing Instructions



- Now go through the migration flow by navigating to /start:
- Go through the domain selection step
- Select the free plan on the Plans page
- On the Goals step, select the "Import existing content or website" option and click continue
- Input the source site URL in the Identify step
- Next, select the "Migrate Site" option on the Import or Migrate step
- Select "Do it for me" on the How to migrate step
- Go through the checkout
- Once you complete the checkout, you should land on the new Credentials step
- Make sure that it looks and functions just like before. Try every case and combination to make sure nothing is broken

## Pre-merge Checklist
